### ISSUE TITLE
PT006 breaks when only one argname

### ISSUE THREAD

**Issue by George-Ogden** [astral-sh/ruff#24715]
### Summary

Pytest has different semantics for single argnames when using a string or a tuple/list.
With a tuple/list, it expects a one-element sequence as each argvalue, so this is valid:
```python
@pytest.mark.parametrize(("number",), [(0,)])
```
However, with a string, it just expects a list of the argvalues, so this is valid:
```python
@pytest.mark.parametrize("number", [0])
```
Rule [PT006](https://docs.astral.sh/ruff/rules/pytest-parametrize-names-wrong-type/#pytest-parametrize-names-wrong-type-pt006) updates the first case to look like this:
```python
@pytest.mark.parametrize("number", [(0,)])
```
which now means that "number" is passed as a tuple and probably breaks the test.

I think the best way to fix this is to flag the error and not fix it when there is one argument.

### Version

ruff 0.15.11

**Comment by ntBre** [comment]
I might be missing something here, but when I put these cases into the [playground](https://play.ruff.rs/b94f3760-0b09-4e0f-8e1b-df80646fe22a), the suggested fix for the first case is:

```python
@pytest.mark.parametrize("number", [0])
def foo(number): ...
```

rather than the tuple output shown in the issue. I also tried various options for [parametrize-names-type](https://docs.astral.sh/ruff/settings/#lint_flake8-pytest-style_parametrize-names-type), and they all produce the output above.

**Comment by George-Ogden** [comment]
Sorry, I simplified my example too much! Here's a case ([in the playground](https://play.ruff.rs/17e87de4-6cca-4493-9111-15ea1af95c13)) that is fixed incorrectly:
```python
@pytest.mark.parametrize(("number",), [(i,)  for i in range(3)])
def foo(number): ...
# becomes
@pytest.mark.parametrize("number", [(i,)  for i in range(3)])
def foo(number): ...
```

**Comment by ntBre** [comment]
Thanks for the update! Hmm, this seems a bit tricky for us to identify. We do have special handling for single-element tuples in lists and tuples but not for list comprehensions. The fix is already unsafe, so it may just be worth documenting this as a fix safety issue rather than adding more special-casing for comprehensions, but I'm curious what others think.

**Comment by George-Ogden** [comment]
Okay, thanks for this response. I'm not familiar with how Ruff is implemented, but is it possible to not fix when this case arises and get the user to do it manually?

**Comment by ntBre** [comment]
If you want to disable the autofix for this rule, you could use the https://docs.astral.sh/ruff/settings/#lint_unfixable option. Something like this:

```toml
[tool.ruff.lint]
unfixable = ["PT006"]
```

**Comment by George-Ogden** [comment]
Thanks, I've added that for the time being. 

> Okay, thanks for this response. I'm not familiar with how Ruff is implemented, but is it possible to not fix when this case arises and get the user to do it manually?

I meant within Ruff. Is there a way to have a rule with partial fixes (auto-fixes except in rare cases)?

**Comment by ntBre** [comment]
Ah yes, sorry. That's kind of what I was getting at with my earlier comment about this being tricky to detect. It's totally fine in Ruff to have fixes that are only provided in certain cases, but we need to be able to reliably identify cases where we should avoid a fix. To do that here, we'd need to recognize the `parametrize` value as a list comprehension and inspect its value to see that it's a tuple. That would be relatively straightforward in this specific case, but we would still miss variations on the same pattern, and the rule could get much more complex trying to handle them all.

**Comment by George-Ogden** [comment]
I think this is actually the special case ([playground](https://play.ruff.rs/8da0d167-7359-49eb-ae05-58bc92cde0f0)):
```python
import pytest

@pytest.mark.parametrize(("number",), [(0,), (1,), (2,)])
def foo(number): ...

@pytest.mark.parametrize(("number",), ((0,), (1,), (2,)))
def foo(number): ...


## fixable cases above (known collection of singletons), unfixable cases below (any other AST node)

@pytest.mark.parametrize(("number",), {(0,), (1,), (2,)}) # it's possible, but unneccessary to fix here
def foo(number): ...

TEST_CASES = [...]
@pytest.mark.parametrize(("number",), TEST_CASES)
def foo(number): ...

def generate_test_case():
    ...

@pytest.mark.parametrize(("number",), (generate_test_case(),))
def foo(number): ...

single_test_case = ...
@pytest.mark.parametrize(("number",), {single_test_case})
def foo(number): ...
```

**Comment by George-Ogden** [comment]
I've also opened #24776 to fix this, as it was a relatively straightforward change. I'm having difficulty building and testing locally because the binaries are so large, so it may take a few iterations to get it correct.
